### ISSUE TITLE
This package clobbers your existing formatter

### ISSUE THREAD

**Issue by hut8** [heroku/rollrus#4]
Hey. I just found this today and it's perfect for what I need, except:

```go
log.SetFormatter(&log.TextFormatter{DisableTimestamp: true})
```

Why is this in here exactly? I'm using `log.JsonFormatter` so I can easily parse my logs on-disk (which contain all sorts of other stuff). I was surprised when a hook actually overwrote this!

**Comment by apg** [comment]
This is here because, internally, our timestamps are generated from [log-shuttle](https://github.com/heroku/log-shuttle). 

This should *definitely* be a configurable. If you are willing, please submit a PR and we'll take a look.

**Comment by hut8** [comment]
Thanks for the quick reply. I will try to get around to it by the weekend.  I think this might just need to be removed if I'm right (since I don't think your package actually uses the formatter :smile: )

Here are my thoughts (your thoughts are much appreciated):

When you get a `logrus.Entry`, which of course is passed to `Fire` which you write, you have exactly the data you want -- `Message`, `Data`, `Time` (which in your case you don't want) and `Level`. Normally, the `logrus.Entry.String()` method will call the `logrus.Logger`'s `Formatter`. Before that, it's just a bunch of fields still. Where are you calling `logrus.Entry.String()`? 

Some of the relevant code that I don't think calls `entry.String()`:
```go
func (r *Hook) Fire(entry *log.Entry) (err error) {
	e := fmt.Errorf(entry.Message)
	m := convertFields(entry.Data)
	if _, exists := m["time"]; !exists {
		m["time"] = entry.Time.Format(time.RFC3339)
	}

	switch entry.Level {
	case log.FatalLevel, log.PanicLevel:
		_, err = r.Client.Critical(e, m)
```

In other words, your `Fire` method already just:
* converts the message to either an `error` (in other words, just gives it an `Error()` method that still doesn't use the `logrus.Formatter`) or leaves it as a string. As far as I can tell, this doesn't ever touch a `logrus.Logger` or `logrus.Formatter`
* calls a function that turns `logrus.Fields` into a `map[string]string`
* passes it to `roll`

**Comment by apg** [comment]
Thanks for the thoughts! You are *probably* entirely correct, and it's a bug. I'll have to do some testing to convince myself of that, of course, before we can merge a PR that removes it. 

Thanks, again!

**Comment by freeformz** [comment]
@hut8 You can now use [`NewHook`](https://godoc.org/github.com/heroku/rollrus#NewHook) (see the Example) to do that on your own log instance.
